src/main/java/com/linkedin/kmf/services/SignalFxMetricsReporterService.java

```diff
@@ -0,0 +1,197 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.linkedin.kmf.services;
+
+import static com.linkedin.kmf.common.Utils.getMBeanAttributeValues;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.MetricRegistry;
+import com.linkedin.kmf.common.MbeanAttributeValue;
+import com.linkedin.kmf.services.configs.SignalFxMetricsReporterServiceConfig;
+import com.signalfx.codahale.metrics.SettableDoubleGauge;
+import com.signalfx.codahale.reporter.MetricMetadata;
+import com.signalfx.codahale.reporter.SignalFxReporter;
+import com.signalfx.endpoint.SignalFxEndpoint;
+
+public class SignalFxMetricsReporterService implements Service {
+  private static final Logger LOG = LoggerFactory.getLogger(SignalFxMetricsReporterService.class);
+
+  private final String _name;
+  private final List<String> _metricNames;
+  private final List<String> _dimensions;
+  private final int _reportIntervalSec;
+  private final ScheduledExecutorService _executor;
+  private final MetricRegistry _metricRegistry;
+  private final SignalFxReporter _signalfxReporter;
+  private final String _signalfxUrl;
+  private final String _signalfxToken;
+  
+  private MetricMetadata _metricMetadata;
+  private Map<String, SettableDoubleGauge> _metricMap; 
+  private Map<String, String> _dimensionsMap;
+  
+  public SignalFxMetricsReporterService(Map<String, Object> props, String name) throws Exception {
+      
+    SignalFxMetricsReporterServiceConfig config = new SignalFxMetricsReporterServiceConfig(props);
+    
+    _name = name;
+    _metricNames = config.getList(SignalFxMetricsReporterServiceConfig.REPORT_METRICS_CONFIG);
+    _dimensions = config.getList(SignalFxMetricsReporterServiceConfig.SIGNALFX_METRIC_DIMENSION);
+    _reportIntervalSec = config.getInt(SignalFxMetricsReporterServiceConfig.REPORT_INTERVAL_SEC_CONFIG);
+    _signalfxUrl = config.getString(SignalFxMetricsReporterServiceConfig.REPORT_SIGNALFX_URL);
+    _signalfxToken = config.getString(SignalFxMetricsReporterServiceConfig.SIGNALFX_TOKEN);
+    
+    if (_signalfxToken.length() < 1) {
+      throw new IllegalArgumentException("SignalFx token is not configured");
+    }
+    
+    _executor = Executors.newSingleThreadScheduledExecutor();
+    _metricRegistry = new MetricRegistry();
+    _metricMap = new HashMap<String, SettableDoubleGauge>();
+    _dimensionsMap = new HashMap<String, String>();
+    setUpDimensionMap();
+    
+    SignalFxReporter.Builder sfxReportBuilder = new SignalFxReporter.Builder(
+        _metricRegistry,
+        _signalfxToken
+    );
+    if (_signalfxUrl.length() > 1) {
```

Comment:
It should probably be an error if the report URL isn't set

Reply:
@mpetazzoni isn't it default to prod url? I want to give this option for our internal use.